pkg/collector/corechecks/servicediscovery/cp_stub.go

```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package servicediscovery
+
+import (
+	"time"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	proccontainers "github.com/DataDog/datadog-agent/pkg/process/util/containers"
+)
+
+type containerProviderStub struct {
+	pidToCid map[int]string
+}
+
+func newContainerProviderStub(targetPIDs []int) proccontainers.ContainerProvider {
+	pidToCid := make(map[int]string)
+
+	for _, pid := range targetPIDs {
+		pidToCid[pid] = "abcd"
+	}
```

Comment:
what is `abcd`?

If that's only a testutility, then please add `test` build flag

Reply:
Added the build flag.

> what is abcd?
Just a dummy Container ID used in the test. I moved it to a const value to sync tests and stub code and prevent mistakes.